WooCommerce/Classes/Analytics/WooAnalytics.swift

```diff
@@ -175,7 +179,7 @@ public extension WooAnalytics {
 //
 extension WooAnalytics {
 
-    func setUserHasOptedOut(_ optedOut: Bool) {
+    public func setUserHasOptedOut(_ optedOut: Bool) {
```

Comment:
The `public` declarations in this file seem strange, if `WooAnalytics` itself is staying in the main target. Nothing should be accessing these from outside the app framework, AFAIK.

Implementing functions of a public protocol doesn't have a requirement that the implementations be public too... they only need to be public if they're declared as public (per function) in the protocol.

I can see that WooAnalytics itself has been public since 2019... and I'm not really sure why that is. Perhaps it started life in another framework?

Reply:
Yup that was a legacy decision that predates my time, and this `public` diff was because of `WooAnalytics` being public. In https://github.com/woocommerce/woocommerce-ios/pull/12685/commits/beaf2d6d0bcf07aecd5bc2b524b8407536d794fb, I updated `WooAnalytics` to internal and no build errors as expected.